services/wallet/walletconnect/database.go

```diff
@@ -2,60 +2,77 @@ package walletconnect
-	Expiry          int64  `json:"expiry"`
-	Active          bool   `json:"active"`
-	DappName        string `json:"dappName"`
-	DappURL         string `json:"dappUrl"`
-	DappDescription string `json:"dappDescription"`
-	DappIcon        string `json:"dappIcon"`
-	DappVerifyURL   string `json:"dappVerifyUrl"`
-	DappPublicKey   string `json:"dappPublicKey"`
+type DBSession struct {
+	Topic            Topic
+	Active           bool
+	SessionJSON      string
+	Expiry           int64
+	CreatedTimestamp int64
+	PairingTopic     Topic
+	IsTest           bool
+	DBDApp
 }
 
-func UpsertSession(db *sql.DB, session DbSession) error {
-	insertSQL := `
-	INSERT OR IGNORE INTO
-		wallet_connect_sessions (topic, pairing_topic, expiry, active)
-	VALUES
-		(?, ?, ?, ?);
-
-	UPDATE
-		wallet_connect_sessions
-	SET
-		pairing_topic = ?,
-		expiry = ?,
-		active = ?,
-		dapp_name = ?,
-		dapp_url = ?,
-		dapp_description = ?,
-		dapp_icon = ?,
-		dapp_verify_url = ?,
-		dapp_publicKey = ?
-	WHERE
-		topic = ?;`
-
-	_, err := db.Exec(insertSQL,
-		session.Topic,
-		session.PairingTopic,
-		session.Expiry,
-		session.Active,
-		session.PairingTopic,
-		session.Expiry,
-		session.Active,
-		session.DappName,
-		session.DappURL,
-		session.DappDescription,
-		session.DappIcon,
-		session.DappVerifyURL,
-		session.DappPublicKey,
-		session.Topic,
-	)
+type DBDApp struct {
+	URL     string
+	Name    string
+	IconURL string
+}
+
+func UpsertSession(db *sql.DB, data DBSession) error {
+	tx, err := db.Begin()
+	if err != nil {
+		return fmt.Errorf("begin transaction: %v", err)
+	}
+	defer func() {
+		if err != nil {
+			tx.Rollback()
+		}
+	}()
+
+	upsertDappStmt := `INSERT INTO wallet_connect_dapps (url, name, icon_url) VALUES (?, ?, ?)
+                   ON CONFLICT(url) DO UPDATE SET name = excluded.name, icon_url = excluded.icon_url`
+	_, err = tx.Exec(upsertDappStmt, data.URL, data.Name, data.IconURL)
+	if err != nil {
+		return fmt.Errorf("upsert wallet_connect_dapps: %v", err)
+	}
+
+	upsertSessionStmt := `INSERT INTO wallet_connect_sessions (
+			topic,
+			active,
+			session_json,
+			expiry,
+			created_timestamp,
+			pairing_topic,
+			is_test,
+			dapp_url
+		)
+		VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+		ON CONFLICT(topic) DO UPDATE SET
+			active = excluded.active,
+			session_json = excluded.session_json,
+			expiry = excluded.expiry,
+			created_timestamp = excluded.created_timestamp,
+			pairing_topic = excluded.pairing_topic,
+			is_test = excluded.is_test,
+			dapp_url = excluded.dapp_url;`
+	_, err = tx.Exec(upsertSessionStmt, data.Topic, data.Active, data.SessionJSON, data.Expiry, data.CreatedTimestamp, data.PairingTopic, data.IsTest, data.URL)
+	if err != nil {
+		return fmt.Errorf("insert session: %v", err)
+	}
+
+	if err = tx.Commit(); err != nil {
+		return fmt.Errorf("commit transaction: %v", err)
+	}
+
+	return nil
+}
+
+func DeleteSession(db *sql.DB, topic Topic) error {
+	_, err := db.Exec("DELETE FROM wallet_connect_sessions WHERE topic = ?", topic)
```

Comment:
![image](https://github.com/status-im/status-go/assets/1925158/d85c5402-bb02-454f-963e-6603f6062f5a)

I'd recommend soft deletes on session. If the business or design comes up with a requirement of session connect history, we'll need this information.

Reply:
Great, will change and mark it as disconnected